### PR TITLE
Expose `PlotLegend` and `ScalarAxis` (axis_y) properties on `TimeSeriesView` blueprint

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Part of [#6083](https://github.com/rerun-io/rerun/issues/6083)

Usage example:
```py
# ...
            # Plot with custom Y range that is locked into place during zoom and legend at the bottom left.
            rrb.TimeSeriesView(
                origin="/plot",
                axis_y=rrb.ScalarAxis(range=[0, 1], lock_range_during_zoom=True),
                plot_legend=rrb.Corner2D.LeftBottom,
            ),
            # Hide plot legend
            rrb.TimeSeriesView(
                origin="/plot",
                plot_legend=rrb.PlotLegend(visible=False),
            ),
# ...
```

Changed codegen type definition:
```fbs
/// A time series view.
table TimeSeriesView (
    "attr.rerun.view_identifier": "TimeSeries"
) {
    /// Configures the vertical axis of the plot.
    axis_y: rerun.blueprint.archetypes.ScalarAxis (order: 1000);

    /// Configures the legend of the plot.
    plot_legend: rerun.blueprint.archetypes.PlotLegend (order: 2000);
}
```

Added the necessary serialization code and snippets. More aggressively using datatypes now to reduce amount of serialization code we need to write in Python.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6114?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6114?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6114)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_types/definitions/rerun/blueprint/components/lock_range_during_zoom.fbs [hunk 1]
```diff
@@ -13,11 +13,10 @@ namespace rerun.blueprint.components;
...
 struct LockRangeDuringZoom (
-    "attr.arrow.transparent",
     "attr.rerun.scope": "blueprint",
-    "attr.rust.derive": "Copy, PartialEq, Eq",
+    "attr.rust.derive": "Copy, Default, PartialEq, Eq, PartialOrd, Ord",
     "attr.rust.repr": "transparent",
     "attr.rust.tuple_struct"
 ) {
-    lock_range: bool (order: 100);
+    lock_range: rerun.datatypes.Bool (order: 100);
```
Comment (emilk) [inline crates/re_types/definitions/rerun/blueprint/components/lock_range_during_zoom.fbs:21]: Why would we sometimes prefer `bool` over `rerun.datatypes.Bool` and the other way around?

File: crates/re_types/src/components/range1d_ext.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    #[inline]
+    pub fn start(&self) -> f64 {
+        self.0 .0[0]
+    }
+
+    #[inline]
+    pub fn end(&self) -> f64 {
+        self.0 .0[1]
+    }
```
Comment (emilk) [inline crates/re_types/src/components/range1d_ext.rs:19]: Do we really want to inherit the terrible names used by the rust ranges? 😬  I would personally prefer more symmetrical names like:

* begin/end
* start/stop
* min/max
* from/to

File: rerun_py/rerun_sdk/rerun/__init__.py [hunk 3]
```diff
@@ -60,42 +97,6 @@
...
-    "script_setup",
-    "script_teardown",
-    "serve",
-    "set_global_data_recording",
-    "set_thread_local_data_recording",
-    "set_time_nanos",
-    "set_time_seconds",
-    "set_time_sequence",
-    "send_blueprint",
-    "spawn",
```
Comment (emilk) [inline rerun_py/rerun_sdk/rerun/__init__.py:98]: I wonder who has the type-sensitive sorter. I guess we'll find out in a PR soon enough

**Review by emilk — APPROVED** [review]
Remember to describe in the PR how a user can set these properties from code. They will want to know when they click the PR link in the changelog!

File: crates/re_types/definitions/rerun/blueprint/components/lock_range_during_zoom.fbs (hunk 1 above)
Comment (Wumpf) [reply crates/re_types/definitions/rerun/blueprint/components/lock_range_during_zoom.fbs:21]: Should always prefer it in Components. We want to get to a place where components always forward to a datatype. Not only will this get us closer to a place where components are a thin semantic layer, it also means I don't have to implement bool serialization again and again in Python (and it won't be generated either for C++, Rust is the odd one out where we regenerate a lot of stuff all the time, to be fixed).
We're not consequent in this yet, but it's already happening e.g. with `PanelExpanded`

File: crates/re_types/src/components/range1d_ext.rs (hunk 2 above)
Comment (Wumpf) [reply crates/re_types/src/components/range1d_ext.rs:19]: 🤷 I have no grudge against start/end. And I'd rather not derive from the pattern established in a standard library if I don't have to

File: rerun_py/rerun_sdk/rerun/__init__.py (hunk 3 above)
Comment (Wumpf) [reply rerun_py/rerun_sdk/rerun/__init__.py:98]: Nope, we won't 😆
* https://github.com/rerun-io/rerun/pull/6117

**Comment by Wumpf** [comment]
added usage example that we can then also use for snippets. Also improved ergonomics to make this possible in the first place

File: crates/re_types/src/components/range1d_ext.rs (hunk 2 above)
Comment (emilk) [reply crates/re_types/src/components/range1d_ext.rs:19]: sadly I guess you are right

**Pull request merged**